src/gcp/cloudfunctions.ts

```diff
@@ -432,9 +432,13 @@ async function list(projectId: string, region: string): Promise<ListFunctionsRes
       unreachable: res.body.unreachable || [],
     };
   } catch (err) {
-    logger.debug("[functions] failed to list functions for " + projectId);
+    logger.debug(`[functions] failed to list functions for ${projectId}`);
     logger.debug(`[functions] ${err?.message}`);
-    return Promise.reject(err?.message);
+    return Promise.reject(
```

Comment:
Indeed, you were right that we shouldn't reject/throw with a string. But since this is an async function you can just `throw` instead of returning a rejected promise.

Reply:
Fixed